frontend/src/main/components/Commons/ManageCows.js

```diff
@@ -7,7 +7,12 @@ const ManageCows = ({ userCommons, commons, onBuy, onSell }) => {
   const [numOfCowsToBuy, setNumOfCowsToBuy] = useState(2);
   const handleBuyHerdModalClose = () => setShowBuyHerdModal(false);
   const handleBuyHerdModalShow = () => setShowBuyHerdModal(true);
-  const handleNumOfCowsToBuyChange = (event) => setNumOfCowsToBuy(parseInt(event.target.value));
+  const handleNumOfCowsToBuyChange = (event) => {
+    const value = parseInt(event.target.value);
+    if (value >= 2 && !isNaN(value)) {
+      setNumOfCowsToBuy(value);
+    }
+  };
```

Comment:
If you want to handle the above in this PR, the check must be done on the server in `UserCommonsController.java`. Doing it in the UI is good from a UX perspective, but it won't protect against bad actors.